flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/StreamingJobGraphGeneratorTest.java

```diff
@@ -654,4 +656,86 @@ public void testNotSupportInputSelectableOperatorIfCheckpointing() {
 
 		StreamingJobGraphGenerator.createJobGraph(env.getStreamGraph());
 	}
+
+	@Test
+	public void testSlotSharingOnAllVerticesInSameSlotSharingGroupByDefaultEnabled() {
+		final StreamGraph streamGraph = createStreamGraphForSlotSharingTest();
+		// specify slot sharing group for map1
+		streamGraph.getStreamNodes().stream()
+			.filter(n -> "map1".equals(n.getOperatorName()))
+			.findFirst()
+			.get()
+			.setSlotSharingGroup("testSlotSharingGroup");
+		streamGraph.getExecutionConfig().enableAllVerticesInSameSlotSharingGroupByDefault();
```

Comment:
Unrelated to this PR but I'm wondering whether a Flink internal setting should be stored in the `ExecutionConfig` which is a user facing structure.

Reply:
If it's for internal use, I think `StreamGraph` is a better place to hold it.
But maybe in the future we will make `allVerticesInSameSlotSharingGroupByDefault` open to users for configurable, so that they can control the strategy according their requirement.

Reply:
Not sure about this tbh. I think in the future I would propose to take a conservative approach here and to not expose things to the user unless really needed.

Reply:
Agreed. David also proposed that in FLINK-14059 just now.
Will move it to StreamGraph in FLINK-14611 latter.